File: server/api/schemas/waitlist.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class WaitlistUser(BaseModel):
    """Упрощенная модель пользователя для waitlist"""

    id: UUID
    first_name: str
    second_name: str
    rank: float
    avatar: Optional[str] = None

    class Config:
        from_attributes = True


class WaitlistResponse(BaseModel):
    id: int
    user_id: UUID
    tournament_id: UUID
    date: datetime
    user: WaitlistUser

    class Config:
        from_attributes = True

File: client/src/types/waitlist.ts
// Упрощенная модель пользователя для waitlist
export type WaitlistUser = {
  id: string // UUID
  first_name: string
  second_name: string
  rank: number
  avatar: string | null
}

export type Waitlist = {
  id: number
  user_id: string // UUID
  tournament_id: string // UUID
  date: string // ISO date string
  user: WaitlistUser
}
